Sources/Fullscreen/NotificationCenter/NotificationCenterView.swift

```diff
@@ -0,0 +1,176 @@
+        tableView.reloadRows(at: indexPaths, with: animated ? .automatic : .none)
+    }
+
+    func resetContentOffset() {
+        guard tableView.numberOfRows(inSection: 0) > 0 else { return }
+
+        tableView.scrollToRow(
+            at: IndexPath(row: 0, section: 0),
+            at: .top,
+            animated: true
+        )
+    }
+}
+
+extension NotificationCenterView: UITableViewDataSource {
+    public func numberOfSections(in tableView: UITableView) -> Int {
+        dataSource?.numberOfSections(in: self) ?? 0
+    }
+
+    public func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        dataSource?.notificationCenterView(self, numberOfRowsIn: section) ?? 0
+    }
+
+    public func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        let model = dataSource?.notificationCenterView(self, modelForRowAt: indexPath)
+        let cell = tableView.dequeue(NotificationCenterCell.self, for: indexPath)
+        let isLastCell = indexPath.row == tableView.numberOfRows(inSection: indexPath.section) - 1
+
+        cell.imageViewDataSource = imageViewDataSource
+
+        cell.configure(
+            with: model,
+            timestamp: delegate?.notificationCenterView(self, timestampForModelAt: indexPath),
+            hideSeparator: isLastCell
+        )
+
+        return cell
+    }
+}
+
+extension NotificationCenterView: UITableViewDelegate {
+    public func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
+        delegate?.notificationCenterView(self, didSelectModelAt: indexPath)
+    }
+
+    public func tableView(_ tableView: UITableView, willDisplay cell: UITableViewCell, forRowAt indexPath: IndexPath) {
+        let lastSection = tableView.numberOfSections - 1
+        let lastRow = tableView.numberOfRows(inSection: lastSection) - 1
+
+        if indexPath.section == lastSection, indexPath.row == lastRow {
+            delegate?.notificationCenterViewWillReachEndOfContent(self)
+        }
+    }
+
+    // Header
+
+    public func tableView(_ tableView: UITableView, viewForHeaderInSection section: Int) -> UIView? {
+        guard let model = delegate?.notificationCenterView(self, modelForSection: section) else { return nil }
+        let header = tableView.dequeue(NotificationCenterSectionHeaderView.self)
+        header.delegate = self
+        header.configure(with: model, inSection: section)
+        return header
+    }
+
+    // Footer
+
+    public func tableView(_ tableView: UITableView, viewForFooterInSection section: Int) -> UIView? {
+        guard section == tableView.numberOfSections - 1 else { return nil }
+        return tableView.dequeue(ActivityIndicatorSectionFooterView.self)
+    }
+
+    public func tableView(_ tableView: UITableView, heightForFooterInSection section: Int) -> CGFloat {
+        guard section == tableView.numberOfSections - 1 else { return .leastNormalMagnitude }
```

Comment:
Since you use this `section == tableView.numberOfSections - 1` a few places here I think it will be easier to read if you pull it out to `isLast(section: section)`